examples/multi-agent-debate-system/agents.py

```diff
@@ -0,0 +1,191 @@
+"""
+agents.py
+
+Defines agents for structured debates. Each function creates a ChatAgent 
+with specific configurations for its assigned role.
+"""
+
+import logging
+import langroid as lr
+from langroid.language_models import OpenAIGPTConfig
+
+# Set up a logger for this module
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def create_agent(base_llm_config: OpenAIGPTConfig, system_message: str) -> lr.ChatAgent:
+    """Creates a ChatAgent with a given system message and configuration.
+
+    Args:
+        base_llm_config (OpenAIGPTConfig): LLM configuration.
+        system_message (str): Role and guidelines for the agent.
+
+    Returns:
+        ChatAgent: Configured chat agent.
+    """
+    logger.info("Creating ChatAgent with system message: %s", system_message[:50])
+    config = lr.ChatAgentConfig(
+        llm=base_llm_config,
+        vecdb=None,
+        system_message=system_message,
+
+    )
+
+    return lr.ChatAgent(config)
+
+
+# Agents for AI in Healthcare
+def create_pro_ai_agent(base_llm_config: OpenAIGPTConfig) -> lr.ChatAgent:
+    """Creates an agent advocating for AI in healthcare."""
+    system_message = """
+    You are an expert in healthcare technology, specializing in AI-driven medical solutions.
+    Your goal is to advocate for the transformative benefits of AI in healthcare while addressing the concerns raised by critics.
+    Focus on the following:
+    1. Highlight specific, compelling examples of how AI improves diagnostics, treatment personalization, operational efficiency, 
+       and healthcare access, introducing fresh examples in each response.
+    2. Engage proactively with opposing arguments. Anticipate concerns such as algorithmic bias, privacy issues, or over-reliance on technology, 
+       and provide detailed rebuttals with real-world evidence and proposed solutions (e.g., federated learning, diverse datasets, explainable AI).
+    3. Acknowledge ethical concerns about AI and argue that these challenges can be mitigated through existing and emerging technologies, 
+       as well as human oversight and regulatory frameworks.
+    4. Avoid redundancy. Build on earlier arguments by introducing new insights, such as AI’s role in global health initiatives, mental health, or medical education.
+
+    Maintain a professional, persuasive, and evidence-based tone, ensuring your arguments address ethical considerations while emphasizing the potential of AI to revolutionize healthcare.
+    Please provide references for all your arguments
```

Comment:
Just a side-note - these references will often be wrong/hallucinated. An idea to think about for later would to enable the agent to do an internet search (using one of the Langroid web search tools) and have it cite a valid reference.